quinn-proto/src/endpoint.rs

```diff
@@ -1052,6 +1050,23 @@ pub enum ConnectError {
     UnsupportedVersion,
 }
 
+/// Error for attempting to retry an [`IncomingConnection`] that can not be retried
+#[derive(Debug, Error)]
+pub struct RetryError(pub IncomingConnection);
+
+impl RetryError {
+    /// Get the [`IncomingConnection`]
+    pub fn into_incoming(self) -> IncomingConnection {
+        self.0
+    }
+}
+
+impl fmt::Display for RetryError {
```

Comment:
Could this be achieved more concisely with an `#[error("...")]` attribute on the type? I think `thiserror` supports that.

Reply:
Done. To both.